securemail/securemail.php

```diff
@@ -6,88 +6,104 @@
+    $enable = intval(get_pconfig(local_user(), 'securemail', 'enable'));
+    $publickey = get_pconfig(local_user(), 'securemail', 'pkey');
 
-    $t = get_markup_template( "admin.tpl", "addon/securemail/" );
+    $t = get_markup_template('admin.tpl', 'addon/securemail/');
 
     $s = replace_macros($t, array(
         '$title' => t('"Secure Mail" Settings'),
         '$submit' => t('Save Settings'),
         '$test' => t('Save and send test'), //NOTE: update also in 'post'
-        '$enable' => array('securemail-enable', t('Enable Secure Mail'), $enable, ""),
-        '$publickey' => array('securemail-pkey', t('Public key'), $publickey, t("Your public PGP key, ascii armored format"), "rows='10'")
+        '$enable' => array('securemail-enable', t('Enable Secure Mail'), $enable, ''),
+        '$publickey' => array('securemail-pkey', t('Public key'), $publickey, t('Your public PGP key, ascii armored format'), 'rows="10"')
     ));
-
-
-
-    return;
 }
-function securemail_settings_post(&$a, &$b){
 
-    if(! local_user()) {
+/**
+ * @brief Handle data from user settings form
+ *
+ * @link https://github.com/friendica/friendica/blob/develop/doc/Plugins.md#plugin_settings_post 'plugin_settings_post' hook
+ *
+ * @param App $a App instance
+ * @param array $b hook data
+ *
+ * @see App
+ */
+function securemail_settings_post(App &$a, array &$b){
+
+    if (!local_user()) {
         return;
     }
 
-    if($_POST['securemail-submit']) {
-        set_pconfig(local_user(),'securemail','pkey',trim($_POST['securemail-pkey']));
-        $enable = ((x($_POST,'securemail-enable')) ? 1 : 0);
-        set_pconfig(local_user(),'securemail','enable', $enable);
-        info( t('Secure Mail Settings saved.') . EOL);
+    if ($_POST['securemail-submit']) {
+        set_pconfig(local_user(), 'securemail', 'pkey', trim($_POST['securemail-pkey']));
+        $enable = ((x($_POST, 'securemail-enable')) ? 1 : 0);
+        set_pconfig(local_user(), 'securemail', 'enable', $enable);
+        info(t('Secure Mail Settings saved.') . EOL);
 
         if ($_POST['securemail-submit'] == t('Save and send test')) {
             $sitename = $a->config['sitename'];
 
             $hostname = $a->get_hostname();
-            if (strpos($hostname, ':')){
+            if (strpos($hostname, ':')) {
                 $hostname = substr($hostname, 0, strpos($hostname, ':'));
             }
 
             $sender_email = $a->config['sender_email'];
-            if (empty($sender_email)){
-                $sender_email = 'noreply@'.$hostname;
+            if (empty($sender_email)) {
+                $sender_email = 'noreply@' . $hostname;
             }
 
-            $subject = "Friendica - Secure Mail - Test";
-            $message = "This is a test message from your Friendica Secure Mail addon.\n\nBye!";
+            $subject = 'Friendica - Secure Mail - Test';
+            $message = 'This is a test message from your Friendica Secure Mail addon.\n\nBye!';
```

Comment:
The two line feeds at the end are lost with the change of quotation style, please use `PHP_EOL` instead.